File: app/src/main/java/com/blackmirror/dongda/ui/activity/enrol/EnrolContract.java
package com.blackmirror.dongda.ui.activity.enrol;


import com.blackmirror.dongda.domain.model.BaseDataBean;
import com.blackmirror.dongda.domain.model.BrandAllLocDomainBean;
import com.blackmirror.dongda.domain.model.LocAllServiceDomainBean;
import com.blackmirror.dongda.ui.BasePresenter;

public class EnrolContract {
    public interface View{

        void onGetBrandAllLocationSuccess(BrandAllLocDomainBean bean);

        void onGetLocAllServiceSuccess(LocAllServiceDomainBean bean);

        void onError(BaseDataBean bean);

    }

    public interface Presenter extends BasePresenter<View> {
        void getBrandAllLocation(String brand_id);
        void getLocAllService(String json, String locations);
    }
}

File: app/src/main/java/com/blackmirror/dongda/ui/activity/enrol/EnrolContract.kt
package com.blackmirror.dongda.ui.activity.enrol


import com.blackmirror.dongda.kdomain.model.BaseDataBean
import com.blackmirror.dongda.kdomain.model.BrandAllLocDomainBean
import com.blackmirror.dongda.kdomain.model.EnrolDomainBean
import com.blackmirror.dongda.kdomain.model.LocAllServiceDomainBean
import com.blackmirror.dongda.ui.BasePresenter

class EnrolContract {

    interface View {

        fun onGetBrandAllLocationSuccess(bean: BrandAllLocDomainBean)

        fun onGetLocAllServiceSuccess(bean: LocAllServiceDomainBean)

        fun onEnrolSuccess(bean: EnrolDomainBean)

        fun onError(bean: BaseDataBean)

    }

    interface Presenter : BasePresenter<View> {
        fun getBrandAllLocation(brand_id: String)
        fun getLocAllService(json: String, locations: String)
        fun enrol(json: String)
    }
}
